shotover-proxy/examples/cassandra_bench.rs

```diff
@@ -0,0 +1,112 @@
+use clap::Parser;
+use test_helpers::docker_compose::DockerCompose;
+use test_helpers::shotover_process::ShotoverProcess;
+
+/// e.g.
+/// cargo run --release --example cassandra_bench -- --example-dir cassandra-passthrough -r 1000
+#[derive(Parser, Clone)]
+#[clap(name = "cassandra_bench", arg_required_else_help = true)]
+pub struct Args {
+    #[clap(short, long)]
+    pub example_dir: String,
+
+    #[clap(short, long)]
+    pub rate: u64,
+}
+
+fn main() {
+    if env!("PROFILE") != "release" {
+        println!("Need to run with --release flag");
+        return;
+    }
+
+    let args = Args::parse();
+
+    // The benches and tests automatically set the working directory to CARGO_MANIFEST_DIR.
+    // We need to do the same as the DockerCompose + ShotoverManager types rely on this.
+    std::env::set_current_dir(env!("CARGO_MANIFEST_DIR")).unwrap();
+
+    let latte = Latte::new(args.rate);
+    {
+        let _compose = DockerCompose::new(&format!(
+            "example-configs/{}/docker-compose.yml",
+            args.example_dir
+        ));
+
+        // Uses ShotoverProcess instead of ShotoverManager for a more accurate benchmark
+        let _shotover_manager = ShotoverProcess::new(&format!(
+            "example-configs/{}/topology.yaml",
+            args.example_dir
+        ));
+
+        println!("Shotover");
+        bench_read(&latte, "localhost:9042");
+        println!("Direct Cassandra");
+        bench_read(&latte, "localhost:9043");
+    }
+
+    println!("Shotover vs Direct Cassandra");
+    latte.compare("read-localhost:9042.json", "read-localhost:9043.json");
+}
+
+fn bench_read(latte: &Latte, address: &str) {
+    latte.bench("read", address)
+}
+
+// TODO: Shelling out directly like this is just for experimenting.
+// Either:
+// * get access to latte as a crate
+// * write our own benchmark logic
+struct Latte {
+    rate: u64,
+}
+
+impl Latte {
+    fn new(rate: u64) -> Latte {
+        test_helpers::docker_compose::run_command(
+            "cargo",
+            &[
+                "install",
+                "--git",
+                "https://github.com/pkolaczk/latte",
+                "--rev",
+                "5f54a4128b76118446de7ccb29223e53ac4e9208",
+            ],
+        )
+        .unwrap();
+        Latte { rate }
+    }
+
+    fn bench(&self, name: &str, address: &str) {
+        std::process::Command::new("latte")
```

Comment:
was it easier to shell out to the bin rather than embedding the app as a dev / bench lib?

Reply:
Its only available as a bin, otherwise I would have used it as a lib